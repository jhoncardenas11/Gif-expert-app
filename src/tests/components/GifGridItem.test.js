import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en el componente GifGridItem', () => {

    const title = "Hola titulo";
    const url = "http://localhost/algo.jpg";
    const wrapper = shallow( <GifGridItem
        url = {url}
        title = {title}
    /> );

    test('Debe mostrar el componente GifGridItem.js de manera correcta', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe tener un parrafo con el title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })

    test('Debe tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })

    test('Debe tener animate__fadeIn', () => {
        const div = wrapper.find('div');
        expect(div.hasClass('animate__fadeIn')).toEqual(true);
    })
    

});