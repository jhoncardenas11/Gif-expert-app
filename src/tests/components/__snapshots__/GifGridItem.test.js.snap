// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGridItem Debe mostrar el componente GifGridItem.js de manera correcta 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Hola titulo"
    src="http://localhost/algo.jpg"
  />
  <p>
    Hola titulo
  </p>
</div>
`;
